### PR TITLE
Add tests for editing the four phone number fields

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
- refactors some existing mocks (mostly renaming) to make them suitable for tests beyond "edit email address" tests
- add tests for adding/editing phone numbers
- [ ] need to add tests for deleting phone numbers

## Testing done
This is all tests :)

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/profile-2/msw-mocks.js [hunk 1]
```diff
@@ -464,20 +450,14 @@ export const editEmailAddressSuccess = () => {
   ];
 };
 
+// Sets up the responses needed to mack a successful email deletion. In
+// particular it mocks the `GET user/` response so that the email address no
+// longer exists
 export const deleteEmailAddressSuccess = [
-  rest.delete(`${prefix}/v0/profile/email_addresses`, (req, res, ctx) => {
+  rest.delete(`${prefix}/v0/profile/*`, (req, res, ctx) => {
```
Comment (sanlouise) [inline src/applications/personalization/profile-2/msw-mocks.js:457]: Do we not need `email_addresses` here?

Comment (sanlouise) [reply src/applications/personalization/profile-2/msw-mocks.js:457]: Or if we want to make this a generic method, at least rename it?

File: src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.edit-telephone.unit.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+  // wait for the edit mode to exit
+  await waitForElementToBeRemoved(phoneNumberInput);
+
+  // the 'edit' button should exist
+  expect(
+    view.getByRole('button', { name: new RegExp(`edit.*${numberName}`, 'i') }),
+  ).to.exist;
+  // and the new number should exist in the DOM
+  // TODO: make better assertions for this?
+  expect(view.getAllByText(newAreaCode, { exact: false }).length).to.eql(4);
```
Comment (sanlouise) [inline src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.edit-telephone.unit.spec.jsx:80]: Learned something new here, `exact: false` "matches substrings and is not case-sensitive" 👍

**Review by sanlouise — APPROVED** [review]
Looks great! Just left a question 😄

File: src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.edit-telephone.unit.spec.jsx (hunk 2 above)
Comment (erikphansen) [reply src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.edit-telephone.unit.spec.jsx:80]: Yeah, I think the alternative here would be something like `getAllByText(new RegExp(newAreaCode))` or maybe even `getAllByText(new RegExp(`${newAreaCode}`))`

File: src/applications/personalization/profile-2/msw-mocks.js (hunk 1 above)
Comment (erikphansen) [reply src/applications/personalization/profile-2/msw-mocks.js:457]: Yeah I should have totally renamed this. Good catch!

Comment (erikphansen) [reply src/applications/personalization/profile-2/msw-mocks.js:457]: Actually this one should keep this name and be kept specific to the `email_addresses` route.

**Review by narin — APPROVED** [review]

**Pull request merged**